modules/ding_wayf_dk/modules/ding_wayf_dk_login/ding_wayf_dk_login.module

```diff
@@ -0,0 +1,255 @@
+      $errors = form_get_errors();
+      if (!empty($errors)) {
+        if (module_exists('ding_wayf_dk_create_user')) {
+          // Clean out message from the login form.
+          drupal_get_messages(NULL, TRUE);
+
+          drupal_set_message(t('You do not have an user account at the public library. If you wish to create an user account !user_url . You are still logged in at WAYF, so you can !logout to logout of WAYF.', array(
+            '!logout' => l(t('click her'), 'wayf/logout'),
+            '!user_url' => l(t('click her'), 'wayf/create-user/before-page'),
+          )), 'warning');
+        }
+        else {
+          // Clean out message from the login form.
+          drupal_get_messages(NULL, TRUE);
+
+          // Set new default message.
+          drupal_set_message(t('Something went wrong. Please try again or !url of WAYF.', array('!url' => l(t('logout'), "wayf/logout"))), 'warning');
+        }
+      }
+    }
+    else {
+      drupal_set_message(t('Something went wrong. Please try again or !url of WAYF.', array('!url' => l(t('logout'), "wayf/logout"))), 'warning');
+    }
+  }
+
+  ding_wayf_dk_login_redirect_user();
+}
+
+/**
+ * Implements hook_wayf_dk_login_pre_authenticate().
+ *
+ * Store destination for the current user, so redirect to e.g. page you where
+ * looking at and not always the front page.
+ */
+function ding_wayf_dk_login_wayf_dk_login_pre_authenticate() {
+  if (isset($_REQUEST['destination'])) {
+    $_SESSION['ding_wayf_dk_login_destination'] = $_REQUEST['destination'];
+
+    // The WAYF authentication process will redirect the users browser to WAYF
+    // servers, so session will not be started. Hence we need to start the
+    // session to ensure that the destination above don't get lost in redirects.
+    drupal_session_start();
+  }
+}
+
+/**
+ * Implements hook_wayf_dk_login_pre_logout().
+ *
+ * Ensures that session is cleaned up.
+ */
+function ding_wayf_dk_login_wayf_dk_login_pre_logout() {
+  if (isset($_SESSION['ding_wayf_dk_login_destination'])) {
+    unset($_SESSION['ding_wayf_dk_login_destination']);
+  }
+}
+
+/**
+ * Implements hook_wayf_dk_login_authenticate_result_alter().
+ *
+ * Restore destination after all the WAYF redirects.
+ */
+function ding_wayf_dk_login_wayf_dk_login_authenticate_result_alter($result) {
+  if (isset($_SESSION['ding_wayf_dk_login_destination'])) {
+    $_REQUEST['destination'] = $_SESSION['ding_wayf_dk_login_destination'];
+  }
+}
+
+/**
+ * Implements hook_preprocess_wayf_dk_login().
+ *
+ * Change the image to Nem-id.
+ */
+function ding_wayf_dk_login_preprocess_wayf_dk_login(&$variables) {
+  $image = array(
+    '#theme' => 'image',
+    '#path' => drupal_get_path('module', 'ding_wayf_dk_login') . '/images/nemid_logo.png',
+    '#alt' => t('Login with NemID'),
+    '#title' => t('Login with NemID'),
+  );
+
+  $variables['login_url']['#text'] = t('Login with') . ' ' . drupal_render($image);
```

Comment:
I appreciate the alt and title here but it does not make sense in relation to the link text which also contains "login with".